### ISSUE TITLE
'Please install pagebotcocoa'

### ISSUE THREAD

**Issue by lsalgueiro** [PageBot/PageBot#198]
Hello, everyone,

I tried installing PageBot, but, upon trying to `import pagebot`, it just yields an error message:
```
Please install pagebotcocoa.
pip install pagebotcocoa
```

… which I also have. Any subsequent installs just read (naturally) 'Requirement already satisfied'…

I'm running Python 3.7.4 in a virtual environment (so everything was supposedly cleanly installed for this purpose) on Mac OS 10.12.6.

Thanks!

**Comment by michielkauwatjoe** [comment]
@lsalgueiro Hi Luís, which pip are you using? Can you enter

which pip

on the command line? Could it be the system-wide pip is called, instead of the one in the virtualenv bin?

**Comment by lsalgueiro** [comment]
@michielkauwatjoe Dear Michael,

so it is! It's pointing to an installation outside the venv — despite the fact that I do seem to have pip installed inside it.

I'm very sorry for the newbish question, then: how do I call the instance inside the venv explicitly?

Thanks!

**Comment by michielkauwatjoe** [comment]
@lsalgueiro depends, for me 

```
pip-3.8
```

(note the dash) works, so check if there's a pip-3.7 in your venv bin folder. It might also be possible to call the command directly using `./path-to`; so from the root of the venv:

```
./bin/pip
```

**Comment by lsalgueiro** [comment]
@michielkauwatjoe I'm very sorry to return to a closed issue, Michael, but this doesn't seem to be the problem after all. Despite `which pip` pointing to the global path, `pagebot` *was* being installed in the virtual env. After calling everything with explicit paths (from pip to the python file), I'm now running up against an error in Vanilla.

```
Either an error occurred while importing pagebotcocoa or it is missing.
Traceback (most recent call last):
  File "/Users/LSalgueiro/Documents/flat/flat/lib/python3.7/site-packages/pagebot/contexts/__init__.py", line 30, in <module>
    from pagebotcocoa.contexts.drawbot.drawbotcontext import DrawBotContext
  File "/Users/LSalgueiro/Documents/flat/flat/lib/python3.7/site-packages/pagebotcocoa/contexts/drawbot/drawbotcontext.py", line 32, in <module>
    from vanilla import *
  File "/Users/LSalgueiro/Documents/flat/flat/lib/python3.7/site-packages/vanilla/__init__.py", line 3, in <module>
    from vanilla.core import Hub
  File "/Users/LSalgueiro/Documents/flat/flat/lib/python3.7/site-packages/vanilla/core.py", line 102
    except Exception, e:
                    ^
SyntaxError: invalid syntax

Please try to (re)install pagebotcocoa.
pip install (--upgrade) pagebotcocoa
```

As you can see, Vanilla is also installed in the virtual env, as far as I can gather.

(the venv is called flat because I was trying to see whether flat was enough for what I was looking for at first, but it makes a lot more sense to go straight to pagebot.)

Thanks a lot in advance for any help you can spare.

**Comment by michielkauwatjoe** [comment]
@lsalgueiro not a problem, that's what issues are for. I think you have the wrong vanilla package. It's a bit confusing but there's another Python package called vanilla. The one we're using is not on PyPi so you need to grab it from github.  We are actually in the process of removing vanilla from the dependencies for this reason, I'll probably upload new versions of pagebot and pagebotcocoa this afternoon. In the meantime, please uninstall vanilla with pip:

```
pip uninstall vanillla
```

and do

```
pip install git+https://github.com/robotools/vanilla.git#egg=vanilla
```

**Comment by lsalgueiro** [comment]
@michielkauwatjoe oops, you were right. I was acquainted with vanilla and aware of this distinction, as it is a dependency for some scripts in another application. However, my first instinct was to do `pip install` before I remembered that I had to do it manually, so I actually wound up with both. 

I finally seem to have everything more or less up and running, but the drawing context for the Hello World seems to raise a `NotImplementedError`… I'll keep on perusing the rest of the examples.
